lisa/sut_orchestrator/azure/platform_.py

```diff
@@ -1377,125 +1377,98 @@ def _parse_detail_errors(self, error: Any) -> List[str]:
 
     # the VM may not be queried after deployed. use retry to mitigate it.
     @retry(exceptions=LisaException, tries=150, delay=2)
-    def _load_vms(
-        self, environment: Environment, log: Logger
-    ) -> Dict[str, VirtualMachine]:
+    def load_vms_info(
```

Comment:
This change makes this method is called multiple times during initialization, it impacts the performance. So it's better to have a separated method to get public ip addresses and just share logics.

Reply:
updated, please review, thanks!